### PR TITLE
Drop interstitial

### PR DESCRIPTION
**Pull request by nikosdouvlis**
## Description

❗ This needs to be merged into #2300 or after #2300 is merged

- Drop interstitial from clerk/backend
- Drop interstitial from clerk/shared
- Drop interstitial from clerk/types
- Drop interstitial from clerk/clerk-js (1 pending item)
- Drop interstitial from clerk/nextjs
- fix build for clerk/remix
- fix build for gatsby-plugin-clerk
- fix build for clerk/fastify
- fix build for clerk/clerk-sdk-node

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [x] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [x] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [x] `@clerk/shared`
- [ ] `@clerk/themes`
- [x] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

**Review by SokratisVidros — APPROVED** [review]

File: packages/nextjs/src/server/authMiddleware.ts [hunk 1]
```diff
@@ -185,40 +186,27 @@ const authMiddleware: AuthMiddleware = (...args: unknown[]) => {
...
-    }
+    const authenticateRequestOptions = {
+      ...options,
+      secretKey: options.secretKey || SECRET_KEY,
+      publishableKey: options.publishableKey || PUBLISHABLE_KEY,
+      ...handleMultiDomainAndProxy(req, options as AuthenticateRequestOptions),
+    } as AuthenticateRequestOptions;
+    const requestState = await clerkClient.authenticateRequest(req, authenticateRequestOptions);
+
+    if (requestState.status === AuthStatus.Handshake) {
```
[the remark this answers is not in the document]
Comment (nikosdouvlis) [reply packages/nextjs/src/server/authMiddleware.ts:197]: Got it, makes sense.
I think we might want to introduce a new status for it or at least think of a different way to transfer this knowledge into the code as it's hard to follow the logic here otherwise

[the remark this answers is not in the document]
Comment (nikosdouvlis) [reply packages/nextjs/src/server/authMiddleware.ts]: Check updated

File: packages/clerk-js/src/core/clerk.ts [hunk 2]
```diff
@@ -1206,9 +1206,10 @@ export class Clerk implements ClerkInterface {
 
   #shouldSyncWithPrimary = (): boolean => {
     if (this.#hasJustSynced()) {
-      if (!this.#options.isInterstitial) {
-        this.#clearJustSynced();
-      }
+      // TODO @nikos
+      // if (!this.#options.isInterstitial) {
+      //   this.#clearJustSynced();
+      // }
```
Comment (BRKalow) [inline packages/clerk-js/src/core/clerk.ts:1212]: I _think_ we can do this:

```suggestion
      this.#clearJustSynced();
```

File: packages/nextjs/src/server/authMiddleware.ts [hunk 3]
```diff
@@ -185,23 +186,25 @@ const authMiddleware: AuthMiddleware = (...args: unknown[]) => {
...
+    } as AuthenticateRequestOptions;
+    const requestState = await clerkClient.authenticateRequest(req, authenticateRequestOptions);
+    const locationHeader = requestState.headers?.get('location');
 
-    const locationHeader = requestStateHeaders?.get('location');
-
-    // triggering a handshake redirect
     if (locationHeader) {
-      return new Response(null, { status: 307, headers: requestStateHeaders });
+      // triggering a handshake redirect
```
Comment (BRKalow) [inline packages/nextjs/src/server/authMiddleware.ts:199]: Okay so I realize that I misspoke earlier...after looking at the logic, we only redirect when the status is handshake, so the check you had initially added is the right approach 👍

**Review by BRKalow — APPROVED** [review]
🤩  so much red 🔪

**Review by dimkl — APPROVED** [review]
😮
